src/qt/bitcoingui.cpp

```diff
@@ -950,6 +956,15 @@ void BitcoinGUI::updateHeadersSyncProgressLabel()
 
 void BitcoinGUI::setNumBlocks(int count, const QDateTime& blockDate, double nVerificationProgress, bool header)
 {
+#ifdef Q_OS_MAC
```

Comment:
Oh my... :D
The original from Bitcoin was already bad...using the ternary operator as a replacement for if/else and at the same time relying on the side effects of the evaluation result... :D
Can you maybe change this to something as simple as:
```
#ifdef Q_OS_MAC
    bool enableAppNap = !masternodeSync.IsSynced();
#ifdef ENABLE_WALLET
    if (privateSendClient.fPrivateSendRunning) {
        enableAppNap = true;
    }
#endif
    if (enableAppNap) {
        m_app_nap_inhibitor->enableAppNap()
    } else {
        m_app_nap_inhibitor->disableAppNap()
    }
#edif
```

Reply:
Updated the code sample to use disableAppNap instead of enableAppNap